### PR TITLE
Refresh token

### PR DESCRIPTION
**Pull request by manojpanta**
Refresh Spotify token

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,8 +1,13 @@
 class SessionsController < ApplicationController
+  cattr_accessor :code
   def create
+    SessionsController.token_code(params[:code])
     party = Party.find_by(identifier: request.env["omniauth.params"]["url"]) if request.env["omniauth.params"]["url"]
     if user = User.find_by(spotify_id: spotify_params(request.env['omniauth.auth'])[:spotify_id])
```
Comment (peregrinebalas) [inline app/controllers/sessions_controller.rb]: Can we change spotify_params to not require an argument like we had discussed during our project check in with Josh yesterday?

Comment (peregrinebalas) [inline app/controllers/sessions_controller.rb]: Why not use User.find_or_create_by(spotify_params) here?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -95,22 +95,36 @@
     DatabaseCleaner.clean
   end
 
+  def stub_oauth_connection
```
Comment (peregrinebalas) [inline spec/rails_helper.rb]: I don't think these methods should be in rails_helper. I missed the PR when this first got put in this file. I'd rather see them in a subdirectory file like `spec/helpers/`.

File: spec/presenters/song_presenter_spec.rb [hunk 3]
```diff
@@ -2,7 +2,15 @@
 
 describe 'SongFacade' do
   before :each do
-    @facade = SongFacade.new('token')
+    user = User.create(spotify_id: "",
```
Comment (peregrinebalas) [inline spec/presenters/song_presenter_spec.rb:5]: This `user` needs to be `@user` as it's being instantiated in each test in this spec.

File: spec/services/spotify_service_spec.rb [hunk 4]
```diff
@@ -2,8 +2,16 @@
 
 describe 'SpotifyClient' do
   before :each do
-    @token = 'token'
-    @service = SpotifyService.new(@token)
+    user = User.create(spotify_id: "31plzrfruxb34tdffhr4vbimuxl4",
```
Comment (peregrinebalas) [inline spec/services/spotify_service_spec.rb:5]: This `user` also needs to be `@user`

Comment (peregrinebalas) [reply spec/services/spotify_service_spec.rb:5]: ^^likely the reason circleci builds are failing.

**Review by peregrinebalas — CHANGES_REQUESTED** [review]

File: spec/presenters/song_presenter_spec.rb (hunk 3 above)
Comment (manojpanta) [reply spec/presenters/song_presenter_spec.rb:5]: We are using @facade for each test not user, so, that should not be an issue

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -1,8 +1,13 @@
...
       session[:user_id] = user.id
+      user.update(access_token: spotify_params(request.env['omniauth.auth'])[:access_token])
+      user.update(refresh_token: spotify_params(request.env['omniauth.auth'])[:refresh_token])
+      user.update(expires_at: spotify_params(request.env['omniauth.auth'])[:expires_at])
```
Comment (peregrinebalas) [inline app/controllers/sessions_controller.rb]: Can we also change expires at to a datetime like we talked about?

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (manojpanta) [reply app/controllers/sessions_controller.rb]: Yeah.Just changed that

Comment (manojpanta) [reply app/controllers/sessions_controller.rb]: Using find or create by now

File: app/controllers/sessions_controller.rb (hunk 5 above)
Comment (manojpanta) [reply app/controllers/sessions_controller.rb]: Removed the column from users table since we are getting new access token off of user.updated_at

**Review by teresa-m-knowles — APPROVED** [review]
Reviewed together, approving

**Pull request merged**